ci/docker_unit_tests.sh

```diff
@@ -1,2 +1,3 @@
 #!/bin/bash
-ci/docker_run.sh logstash-unit-tests ci/unit_tests.sh $@
+SONAR_TOKEN=$(vault read -field=token secret/ci/elastic-logstash/sonar-creds)
```

Comment:
Judging by the test failures, it looks like there is an issue pointing to the correct vault:

```
09:08:37 Error reading secret/ci/elastic-logstash/sonar-creds: Get "https://127.0.0.1:8200/v1/secret/ci/elastic-logstash/sonar-creds": dial tcp 127.0.0.1:8200: connect: connection refused
``

I think you may need to pass the vault environment variables into the jenkins jobs, probably doing something along the lines of https://github.com/elastic/infra/blob/9e0feddfb4d7409342734de4c8f247ea5c96b6cb/ci/jjb/logstash-ci/defs/elastic-logstash/on-demand.yml#L34 for the Logstash jobs defined [here](https://github.com/elastic/infra/tree/9e0feddfb4d7409342734de4c8f247ea5c96b6cb/ci/jjb/logstash-ci/defs/elastic-logstash)